tools/wptrunner/wptrunner/update/tree.py

```diff
@@ -178,11 +185,27 @@ def add_new(self, prefix=None):
                        add all files under that path.
         """
         if prefix is None:
-            args = ("-a",)
+            args = ["-a"]
         else:
-            args = ("--no-ignore-removal", prefix)
+            args = ["--no-ignore-removal", prefix]
         self.git("add", *args)
 
+    def add_ignored(self, sync_tree, prefix):
+        """Add files to the staging area that are explicitly ignored by git.
+
+        :param prefix: None to include all files or a path prefix to
+                       add all files under that path.
+        """
+        with tempfile.TemporaryFile() as f:
```

Comment:
Using a temp file seems potentially unnecessary? You could pipe the commands together, but it might be simpler to use a `io.BytesIO` here? Up to you.

Reply:
io.BytesIO does not appear to be supported as a stdin or stdout argument for subprocess.check_output/subprocess.check_call in Python <3.0. Using subprocess.PIPE with these functions is called out as dangerous by the docs; I would rather stick with the implementation that exists and works than continue to try to find something else that I can use instead.